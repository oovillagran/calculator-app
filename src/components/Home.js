import '../styles/Home.css';

export default function Home() {
  return (
    <div className="home">
      <h1 className="home-title">Welcome to our page!</h1>
      <p className="home-text">
        Hello Friends! &quot;Math magicians&quot; is a website for all fans of mathematics.
        It is a Single Page App (SPA) that allows users to:
      </p>
      <ul className="lists">
        <li>
          Make simple calculations.
        </li>
        <li>
          Read a random math-related quote.
        </li>
      </ul>
      <p className="home-text">
        By building this application, I learned how to use React.
        <br />
        This application includes three subpages:
      </p>
      <ul>
        <li>
          Home Page.
        </li>
        <li>
          Calculator Page.
        </li>
        <li>
          Quote Page.
        </li>
      </ul>
      <p className="home-text">
        By building this application I was able to further strengthen my knowledge in JavaScript and
        had the opportunity to start working using React.
        <br />
        <br />
        React is a really wonderfull library that allows developers working in a more efficient way.
        <br />
        Among the main features used for the creation of this application are:
      </p>
      <ul>
        <li>
          Setting up the environment and tools needed to develop a React application.
        </li>
        <li>
          Develop React components.
        </li>
        <li>
          Connect to an API within a functional component.
        </li>
      </ul>
    </div>
  );
}
