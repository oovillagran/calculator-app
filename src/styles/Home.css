.home {
  margin: 1rem auto;
  padding: 0.75rem;
  color: #fff;
  width: 90%;
  line-height: 1.5rem;
}

.home-title {
  text-align: center;
  padding: 2rem;
}

@media only screen and (max-width: 768px) {
  .home-title {
    padding: 0.5rem;
  }

  .home {
    margin: 0.25rem;
    line-height: 1rem;
    font-size: 0.75rem;
  }
}
