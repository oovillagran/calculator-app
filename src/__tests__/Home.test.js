import React from 'react';
import { render } from '@testing-library/react';
import Home from '../components/Home';
import '@testing-library/jest-dom/extend-expect';

describe('Home component', () => {
  test('renders the welcome message', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
