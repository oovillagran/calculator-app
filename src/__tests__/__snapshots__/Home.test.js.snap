// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders the welcome message 1`] = `
<div>
  <div
    class="home"
  >
    <h1
      class="home-title"
    >
      Welcome to our page!
    </h1>
    <p
      class="home-text"
    >
      Hello Friends! "Math magicians" is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to:
    </p>
    <ul
      class="lists"
    >
      <li>
        Make simple calculations.
      </li>
      <li>
        Read a random math-related quote.
      </li>
    </ul>
    <p
      class="home-text"
    >
      By building this application, I learned how to use React.
      <br />
      This application includes three subpages:
    </p>
    <ul>
      <li>
        Home Page.
      </li>
      <li>
        Calculator Page.
      </li>
      <li>
        Quote Page.
      </li>
    </ul>
    <p
      class="home-text"
    >
      By building this application I was able to further strengthen my knowledge in JavaScript and had the opportunity to start working using React.
      <br />
      <br />
      React is a really wonderfull library that allows developers working in a more efficient way.
      <br />
      Among the main features used for the creation of this application are:
    </p>
    <ul>
      <li>
        Setting up the environment and tools needed to develop a React application.
      </li>
      <li>
        Develop React components.
      </li>
      <li>
        Connect to an API within a functional component.
      </li>
    </ul>
  </div>
</div>
`;
